flyteplugins/go/tasks/plugins/k8s/batchscheduler/scheduler/yunikorn/yunikorn.go

```diff
@@ -0,0 +1,34 @@
+package yunikorn
+
+import (
+	"context"
+	"github.com/flyteorg/flyte/flyteplugins/go/tasks/plugins/k8s/batchscheduler"
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	Yunikorn            = "yunikorn"
+	AppID               = "yunikorn.apache.org/app-id"
+	TaskGroupNameKey    = "yunikorn.apache.org/task-group-name"
+	TaskGroupsKey       = "yunikorn.apache.org/task-groups"
+	TaskGroupParameters = "yunikorn.apache.org/schedulingPolicyParameters"
+)
+
+type YunikornSchedulerManager struct {
+	parameters string
+}
+
+func (y *YunikornSchedulerManager) Mutate(ctx context.Context, object client.Object) error {
+	switch object.(type) {
+	case *rayv1.RayJob:
```

Comment:
this is weird isnt it? 
WE have a common interface, but have a switch for each?
In this case, we should allow this overriding to exist next to ray job plugin and not with yunikorn. this will break when things update pretty quickly.

Why have a common interface?